lib/finch/http1/conn.ex

```diff
@@ -223,20 +283,27 @@ defmodule Finch.Conn do
               mint,
               ref,
               timeout,
-              value,
-              headers
+              fields,
+              %{resp_metadata | status: value}
             )
 
           {:halt, acc} ->
             {:ok, mint} = Mint.HTTP1.close(mint)
-            {:ok, mint, acc, {status, headers}}
+            {:ok, mint, acc, resp_metadata}
 
           other ->
             raise ArgumentError, "expected {:cont, acc} or {:halt, acc}, got: #{inspect(other)}"
         end
 
       {:headers, ^ref, value} ->
-        case fun.({:headers, value}, acc) do
+        resp_metadata =
+          if fields == :headers do
+            %{resp_metadata | headers: value}
+          else
+            %{resp_metadata | trailers: value}
```

Comment:
isn't it possible that we could get multiple :headers or :trailers messages, meaning we need to concat `value` to the value we have in the accumulator?